### DISCUSSION TITLE
Multi-state modelling?

### DISCUSSION THREAD

**Discussion by terryfrankcombe** [mir-group/nequip#347] [General]
What are the prospects to get NequIP to fit to multi-state potential energy matrices?

Ideally one would train a model to return a matrix.  Another approach might be to separately train models for each adiabatic surface.  I guess with care one may be able to track adiabats through conical intersections to avoid discontinuous derivatives.  Avoided crossings, with high derivatives, might be challenging.  Once could also train a coupling surface.  But this approach would give divergent coupling along CI seams.

Would you expect NequIP to be able to handle such surfaces?  How easy would it be to set up to train matrices?

**Comment by Linux-cpp-lisp** [comment]
Hi @terryfrankcombe ,

Interesting question--- the most relevant piece of work I know (although I am not familiar with the details) is this:

> Axelrod, S., Shakhnovich, E. & Gómez-Bombarelli, R. Excited state non-adiabatic dynamics of large photoswitchable molecules using a chemically transferable machine learning potential. Nat Commun 13, 3440 (2022). https://doi.org/10.1038/s41467-022-30999-w

https://www.nature.com/articles/s41467-022-30999-w

The methods section seems to describe an output block and set of loss functions put on top of an existing neural network architecture treated more or less as a black box.  Adding such extensions to NequIP/Allegro is certainly possible, though it is always important to carefully consider the normalization and extensivity of any new property prediction or other things you add: https://github.com/mir-group/nequip-example-extension.
